cql3/expr/expression.hh

```diff
@@ -540,6 +540,21 @@ const binary_operator* find_binop(const expression& e, Fn predicate_fun) {
     return find_in_expression<binary_operator>(e, predicate_fun);
 }
 
+// Goes over each expression of the specified type and calls for_each_func for each of them.
+// For example:
+// for_each_expression<column_vaue>(e, [](const column_value& cval) {std::cout << cval << '\n';});
+// Will print all column values in an expression
+template<ExpressionElement ExprElem, class Fn>
+requires std::invocable<Fn, const ExprElem&>
+void for_each_expression(const expression& e, Fn for_each_func) {
```

Comment:
I believe that in the long term this will be less useful, because selecting on just the type is not enough context.

It's okay for now where the grammar imposes severe restrictions on where expression subtypes can exist.